yt_dlp/extractor/xumo.py

```diff
@@ -0,0 +1,155 @@
+            'description': 'An academy coed (Virginia Madsen) discovers a faculty plot to achieve immortality with a serum made from students\' brains.',
+            'duration': 5436,
+            'release_year': 1987,
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'ext': 'mp4'
+        }
+    }, {
+        # series
+        'url': 'https://www.xumo.tv/tv-shows/super-mario-world/XM0AN69OG47PRN',
+        'playlist_count': 10,
+        'info_dict': {
+            'id': 'XM0AN69OG47PRN',
+            'title': 'Super Mario World'
+        },
+        'params': {
+            'skip_download': True
+        }
+    }]
+
+    _INFO_URL = 'https://valencia-app-mds.xumo.com/v2/assets/asset/'
+    _INFO_QUERY_PARAMS = {
+        'f': ['connectorId',
+              'title',
+              'providers',
+              'descriptions',
+              'runtime',
+              'originalReleaseYear',
+              'cuePoints',
+              'ratings',
+              'hasCaptions',
+              'availableSince',
+              'genres',
+              'season:all',
+              'episodes.episodeTitle',
+              'episodes.runtime',
+              'episodes.descriptions',
+              'episodes.hasCaptions',
+              'episodes.ratings']
+    }
+
+    def _get_video_info(self, video_id, info_json):
+        formats, subtitles = [], {}
+        has_drm = False
+        for provider in info_json.get('providers') or []:
+            for source in provider.get('sources') or []:
+
+                fmts, subs = [], {}
+
+                format_url = url_or_none(source.get('uri'))
+                if not format_url:
+                    continue
+                ext = determine_ext(format_url)
+                if ext == 'm3u8':
+                    fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                        format_url, video_id, entry_protocol='m3u8_native',
+                        m3u8_id='hls', fatal=False)
+                elif ext == 'mpd':
+                    fmts, subs = self._extract_mpd_formats_and_subtitles(
+                        format_url, video_id, mpd_id='dash', fatal=False)
+                elif format_url.endswith('.ism/Manifest'):
+                    fmts, subs = self._extract_ism_formats_and_subtitles(
+                        format_url, video_id, ism_id='mss', fatal=False)
+
+                formats.extend(fmts)
+                subtitles = self._merge_subtitles(subtitles, subs)
+
+            for caption in provider.get('captions') or []:
+                sub_url = caption.get('url')
+                ext = determine_ext(sub_url)
+                if ext == 'vtt' or ext == 'srt':
+                    subs = {
+                        (caption.get('lang') or ''): [{
+                            'ext': determine_ext(sub_url),
+                            'url': sub_url
+                        }]
+                    }
+                    subtitles = self._merge_subtitles(subtitles, subs)
```

Comment:
```suggestion
                if ext not in ('vtt', 'srt'):
                    continue
                subtitles.setdefault(caption.get('lang') or 'und', []).append({
                    'ext': ext, 'url': sub_url
                })
```

Reply:
Didn't see this earlier since Github was hiding it. Just committed it manually.